src/privatebin.rs

```diff
@@ -36,25 +38,70 @@ pub struct Paste {
     pub ct: String,
     pub meta: Meta,
     pub adata: Data,
+    pub comments: Option<Vec<Comment>>,
 
     #[serde(skip)]
     pub adata_str: String,
 }
 
+impl<'a> Decryptable<'a> for Paste {
+    fn get_ct(&'a self) -> &'a str {
+        &self.ct
+    }
+    fn get_cipher(&'a self) -> &'a Cipher {
+        &self.adata.cipher
+    }
+    fn get_adata_str(&self) -> String {
+        self.adata_str.clone()
+    }
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+pub struct Comment {
+    pub status: Option<i32>,
+    pub id: String,
+    pub pasteid: String,
+    pub parentid: String,
+    pub url: Option<String>,
+    pub v: i32,
+    pub ct: String,
+    pub meta: Meta,
+    pub adata: Cipher,
+
+    #[serde(skip)]
+    pub adata_str: String,
+}
+
+impl<'a> Decryptable<'a> for Comment {
+    fn get_ct(&'a self) -> &'a str {
+        &self.ct
+    }
+    fn get_cipher(&'a self) -> &'a Cipher {
+        &self.adata
+    }
+    fn get_adata_str(&self) -> String {
+        if self.adata_str.is_empty() {
+            serde_json::to_string(&self.adata).unwrap()
+        } else {
+            self.adata_str.clone()
+        }
+    }
+}
+
 #[derive(Deserialize, Debug, Serialize)]
 pub struct Meta {
     pub created: i32,
 }
 
-#[derive(Deserialize, Debug, Serialize)]
+#[derive(Deserialize, Debug)]
 pub struct Data {
     pub cipher: Cipher,
     pub format: PasteFormat,
     pub discuss: i8,
     pub burn: i8,
 }
 
-#[derive(Deserialize, Debug, Serialize)]
+#[derive(Deserialize, Debug)]
```

Comment:
(same question as above) Why did we remove Serialize here and implement it ourselves? If the reason is identical just resolve this one, to keep the discussion in one place.

Reply:
Yes, the reasons are same.